tests/spec/s-array.js

```diff
@@ -1568,4 +1568,141 @@ describe('Array', function () {
             expect(result).toEqual(['2', '3']);
         });
     });
+    describe('#reverse()', function () {
+        it('works with array', function () {
+            var arrCmp = [
+                undefined,
+                null,
+                -1,
+                0,
+                1,
+                false,
+                true,
+                undefined,
+                '',
+                'abc',
+                null,
+                undefined
+            ],
+            expected = [
+                undefined,
+                null,
+                'abc',
+                '',
+                undefined,
+                true,
+                false,
+                1,
+                0,
+                -1,
+                null,
+                undefined
+            ];
+            arrCmp.reverse();
+            expect(arrCmp).toEqual(expected);
+        });
+        it('works with arguments', function () {
+            var arrCmp = (function () {
+                return arguments;
+            }(
+                undefined,
+                null,
+                -1,
+                0,
+                1,
+                false,
+                true,
+                undefined,
+                '',
+                'abc',
+                null,
+                undefined
+            ));
+            var expected = (function () {
+                return arguments;
+            }(
+                undefined,
+                null,
+                'abc',
+                '',
+                undefined,
+                true,
+                false,
+                1,
+                0,
+                -1,
+                null,
+                undefined
+            ));
+            Array.prototype.reverse.call(arrCmp);
+            expect(Object.prototype.toString.call(arrCmp)).toBe('[object Arguments]');
```

Comment:
With the `call` fix this would work reliably (well, as reliable as duck-typing is), but I can change this to an `isArguments` function.